tapir/shifts/views.py

```diff
@@ -638,3 +642,55 @@ def get_context_data(self, **kwargs):
 
     def get_success_url(self):
         return self.get_target_user().get_absolute_url()
+
+
+class CreateShiftExemptionView(PermissionRequiredMixin, CreateView):
+    model = ShiftExemption
+    form_class = ShiftExemptionForm
+    permission_required = "shifts.manage"
+
+    def get_target_user(self) -> TapirUser:
+        return ShiftUserData.objects.get(pk=self.kwargs["shift_user_data_pk"]).user
+
+    def form_valid(self, form):
+        form.instance.user = self.get_target_user()
+        form.instance.shift_user_data = form.instance.user.shift_user_data
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        context["user"] = self.get_target_user()
```

Comment:
I'm always a bit uneasy about using `user` because it clashes with the built-in variable in the Django auth system. But I haven't really found a much better way of doing things. Sometimes, I do it like this, sometimes I use `selected_user`. But I don't really have a good solution either, I feel ideally the built-in user should not be there.